MayssaraOmar/assignment3/Word search/src/Word search.cpp

```diff
@@ -0,0 +1,48 @@
+#include <map>
+#include "word search.h"
+using namespace std;
+
+void get_words_from_grid(vector<vector<char> > &grid, int row, int col,
+		string word, map<pair<int, int>, bool> &vis,
+		map<pair<pair<int, int>, string>, bool> &memo, dictionary &dic,
+		unordered_set<string> &words) {
+	if (row < 0 || row >= grid.size() || col < 0 || col >= grid[row].size()) {
+		return;
+	}
+	if (!dic.isPrefix(word + grid[row][col])
+			|| memo[make_pair(make_pair(row, col), word)]) {
+		return;
+	}
+
+	if (dic.isWord(word + grid[row][col])) {
```

Comment:
This depends on exactly how `isPrefix()` works, but it is possible it would not return true for a completed word that is a prefix of another word. In that case, I would suggest checked `isWord()` first.